### PR TITLE
Update VCPKG Baseline due to zlib

### LINKED ISSUE #4026: VCPKG ZLIB version obsolete
**Describe the bug**
VCPKG version of ZLIB is 1.2.12 the publisher removed this version in favor of 1.12.13 due to a security issue in 1.2.12 https://nvd.nist.gov/vuln/detail/CVE-2022-37434 
https://zlib.net/ 
We need to get ZLIB publisher to update the version. 

***Exception or Stack Trace***
Add the exception log and stack trace if available

**To Reproduce**
>vcpkg install zlib

Error: Failed to download from mirror set:
https://www.zlib.net/zlib-1.2.12.tar.gz

> curl https://www.zlib.net/zlib-1.2.12.tar.gz

<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>404 Not Found</title>

***Code Snippet***
Add the code snippet that causes the issue.

**Expected behavior**

**Screenshots**

**Setup (please complete the following information):**
 - IDE : [e.g. IntelliJ]
 - Version of the Library used

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [ ] Bug Description Added
- [ ] Repro Steps Added
- [ ] Setup information Added

### PR DESCRIPTION
**Pull request by gearama**
override so it picks up from cached location
closes #4026 
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Review by vhvb1989 — APPROVED** [review]

File: vcpkg.json [hunk 1]
```diff
@@ -22,5 +22,11 @@
       "name": "wil",
       "platform": "windows"
     }
+  ],
+  "overrides": [
+    {
+      "name": "zlib",
+      "version": "1.2.12#2"
```
Comment (Jinming-Hu) [inline vcpkg.json:29]: I noticed the latest version on vcpkg is still 1.2.12. https://github.com/microsoft/vcpkg/blob/06b5f4a769d848d1a20fa0acd556019728b56273/ports/zlib/vcpkg.json#L3

How do we build the repo locally with `1.2.12#2`?

Comment (LarryOsterman) [reply vcpkg.json:29]: If you're using a current vcpkg.exe, then the overrides should resolve the problem. Sync your vcpkg repo to the current master branch and I believe it should resolve the issue.

**Pull request merged**